src/json/json_interface.cpp

```diff
@@ -0,0 +1,101 @@
+/*******************************************************************\
+
+Module: JSON Commandline Interface
+
+Author: Peter Schrammel
+
+\*******************************************************************/
+
+/// \file
+/// JSON Commandline Interface
+
+#include "json_interface.h"
+
+#include <iostream>
+
+#include <util/cmdline.h>
+#include <util/exception_utils.h>
+#include <util/json.h>
+#include <util/message.h>
+
+#include "json_parser.h"
+
+/// Parse commandline options from \p json into \p cmdline
+static void get_json_options(const json_objectt &json, cmdlinet &cmdline)
+{
+  const jsont &arguments = json["arguments"];
+  if(!arguments.is_array())
+  {
+    throw invalid_command_line_argument_exceptiont(
+      "array expected", "'arguments'");
+  }
+
+  for(const auto &argument : to_json_array(arguments))
+  {
+    if(argument.is_string())
+      cmdline.args.push_back(argument.value);
+    else
+    {
+      throw invalid_command_line_argument_exceptiont(
+        "string expected", "argument");
+    }
+  }
+
+  const jsont &options = json["options"];
+  if(!options.is_object())
+  {
+    throw invalid_command_line_argument_exceptiont(
+      "array expected", "'options'");
+  }
+  for(const auto &option_pair : to_json_object(options))
```

Comment:
Nit pick: add a blank line before this one to be consistent with the code a few lines above.